tutorials/01-Introduction.ipynb

```diff
@@ -0,0 +1,2872 @@
+     "execution_count": 62,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "fibonacci(1001)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7ab4f831-7bed-42f4-9ba9-ac4b2689405d",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "#### Prime Factorization"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "97ec23e0-7437-4235-98fb-c4e374266a97",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "factorint(1234567890123456789)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "16111013-fb7f-476e-9cf4-967667627e01",
+   "metadata": {
+    "editable": false,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "## Rational Numbers"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "40bbdd9b-3cbb-43a6-9a59-b1cf43cf9a43",
+   "metadata": {},
+   "source": [
+    "### Why floating points are bad?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fa4708dc-27c9-4b20-841c-ba18d0c3a323",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "Why does `0.1 + 0.2` does not equal `0.3` in Python?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 42,
+   "id": "2bbcc875-c488-4b59-9e36-a0dbea9d0428",
+   "metadata": {
+    "editable": false,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "False"
+      ]
+     },
+     "execution_count": 42,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "0.1 + 0.2 == 0.3"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "82552901-7f3d-4895-b5de-283b2a12fff4",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "This is because the numbers are not exactly represented as `0.1, 0.2, 0.3` in computer \n",
+    "and tiny error term appears in subtraction."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 41,
+   "id": "b4a4fda1-8f8f-44a7-97ee-d961abed986c",
+   "metadata": {
+    "editable": false,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "5.551115123125783e-17"
+      ]
+     },
+     "execution_count": 41,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "0.1 + 0.2 - 0.3"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ea1c8e7a-3aa2-4fe0-8449-2344ac89b2f4",
+   "metadata": {},
+   "source": [
+    "The floating point numbers are very inaccurate objects use in mathematics, or in business.\n",
```

Comment:
Again, floating-point numbers are not inherently bad. They are used all over the place, and it's just wrong to call them "very inaccurate". The point we should be making here is that SymPy itself is designed around exact mathematics, and floats are a bad fit for that design. Plus, if you do want to use floats, and you are only doing pure numeric work, you are better off using a purely numeric library like NumPy.